homeassistant/components/sensor/mqtt.py

```diff
@@ -74,15 +82,36 @@ def async_added_to_hass(self):
         @callback
         def message_received(topic, payload, qos):
             """A new MQTT message has been received."""
+            # auto-expire enabled?
+            if self._expire_after > 0:
+                # Reset expiration time and set trigger
+                self._value_expiration_at = (
+                    dt_util.utcnow() +
+                    timedelta(seconds=self._expire_after)
+                    )
+                async_track_point_in_utc_time(self.hass,
```

Comment:
To answer this question, the mqtt component is written using Python's asyncio library, which means that only one of these callbacks can be running at a time. If you do some searching for asyncio you can find some good literature.